pkgs/development/compilers/flutter/patches/flutter310/move-cache.patch

```diff
@@ -0,0 +1,152 @@
+diff --git a/packages/flutter_tools/bin/podhelper.rb b/packages/flutter_tools/bin/podhelper.rb
+index 01d0bf4e21..cb00a44674 100644
+--- a/packages/flutter_tools/bin/podhelper.rb
++++ b/packages/flutter_tools/bin/podhelper.rb
+@@ -39,7 +39,7 @@ def flutter_additional_ios_build_settings(target)
+
+   # This podhelper script is at $FLUTTER_ROOT/packages/flutter_tools/bin.
+   # Add search paths from $FLUTTER_ROOT/bin/cache/artifacts/engine.
+-  artifacts_dir = File.join('..', '..', '..', '..', 'bin', 'cache', 'artifacts', 'engine')
++  artifacts_dir = File.join(ENV['HOME'], '.cache/flutter/artifacts/engine')
```

Comment:
Does darwin use .cache directory in HOME for artifacts? Currently Flutter in nixpkgs only uses .cache for the lockfile and everything else is readonly in `<flutter>/bin/cache`

Reply:
This PR is based on https://github.com/NixOS/nixpkgs/pull/196133, so not sure about this 🤔